File: skylot222_ui/src/services/WishlistService.js
import http from "../http-common"; 

class WishlistService {
  getAllWishlists(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/wishlist/wishlists`, searchDTO);
  }

  get(wishlistId) {
    return this.getRequest(`/wishlist/${wishlistId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/wishlist?field=${matchData}`, null);
  }

  addWishlist(data) {
    return http.post("/wishlist/addWishlist", data);
  }

  update(data) {
  	return http.post("/wishlist/updateWishlist", data);
  }
  
  uploadImage(data,wishlistId) {
  	return http.postForm("/wishlist/uploadImage/"+wishlistId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new WishlistService();

File: src/main/java/com/skylot222/service/WishlistService.java
package com.skylot222.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylot222.domain.Wishlist;
import com.skylot222.dto.WishlistDTO;
import com.skylot222.dto.WishlistSearchDTO;
import com.skylot222.dto.WishlistPageDTO;
import com.skylot222.dto.WishlistConvertCriteriaDTO;
import com.skylot222.service.GenericService;
import com.skylot222.dto.common.RequestDTO;
import com.skylot222.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface WishlistService extends GenericService<Wishlist, Integer> {

	List<Wishlist> findAll();

	ResultDTO addWishlist(WishlistDTO wishlistDTO, RequestDTO requestDTO);

	ResultDTO updateWishlist(WishlistDTO wishlistDTO, RequestDTO requestDTO);

    Page<Wishlist> getAllWishlists(Pageable pageable);

    Page<Wishlist> getAllWishlists(Specification<Wishlist> spec, Pageable pageable);

	ResponseEntity<WishlistPageDTO> getWishlists(WishlistSearchDTO wishlistSearchDTO);
	
	List<WishlistDTO> convertWishlistsToWishlistDTOs(List<Wishlist> wishlists, WishlistConvertCriteriaDTO convertCriteria);

	WishlistDTO getWishlistDTOById(Integer wishlistId);







}





